File: src/main/java/com/richard/walker/outlets/JamaicanOutlet.java
package outlets.java;

public class JamaicanOutlet implements NorthAmericanOutlet{
    private static final double WORKING_VOLTAGE = 110.00;

    @Override
    public Double provide110Volts(){
        System.out.println("Jamaican Outlet providing " + WORKING_VOLTAGE + " of power!");
        return WORKING_VOLTAGE;
    }

    @Override
    public Boolean isFunctioning(){
        System.out.println("Jamaican Outlet is working.");
        if(WORKING_VOLTAGE >= (WORKING_VOLTAGE * 0.90)){
            return Boolean.TRUE;
        } else{
            return Boolean.FALSE;
        }
    }
}

File: src/com/richard/walker/outlets/kotlin/JamaicanOutlet.kt
package com.richard.walker.outlets.kotlin

private const val WORKING_VOLTAGE = 110.0

class JamaicanOutlet: NorthAmericanOutlet {
    override fun provide110Volts(): Double {
        println("Jamaican outlet providing $WORKING_VOLTAGE V of power!")
        return WORKING_VOLTAGE
    }

    override fun isFunctioning(): Boolean {
        val isVoltageAcceptable = WORKING_VOLTAGE >= WORKING_VOLTAGE * .90

        if (isVoltageAcceptable)
            println("British outlet is working")
        else
            println("British outlet is not working")

        return isVoltageAcceptable
    }

}